packages/vscode-extension/src/project/project.ts

```diff
@@ -671,7 +671,8 @@ export class Project
   private checkIfNativeChanged = throttleAsync(async () => {
     if (!this.isCachedBuildStale && this.projectState.selectedDevice) {
       const platform = this.projectState.selectedDevice.platform;
-      const isCacheStale = await PlatformBuildCache.forPlatform(platform).isCacheStale();
+      const buildCache = new BuildCache(platform, getAppRootFolder());
```

Comment:
would it be possible to restructure the code in a way that we don't have to instantiate the new cache object here? This doesn't matter at this point because the class is stateless, but may be a source of errors in the future if we end up changing that.